wordpress-mobile/WordPress-iOS
# Notifications split view: immediately remove deleted comment notifications

Ref: #17790

In split view, when a comment has been permanently deleted, the Notifications list is immediately updated. The Comment Notification is removed, and the first available Notification is selected.

Also, since there is sometimes a delay between tapping the `Delete Permanently` button and the comment actually being deleted, the button now shows an activity indicator. This is in any comment detail view.

Implementation details:
Since `NotificationCommentDetailCoordinator` could be init'ed from either Notifications or Notification Details, using a protocol/delegate would be a bit messy. So I opted for a simple `NSNotification` that Notifications can just listen for.

To test:
- In split view, go to Notifications. 
- Select a Comment Notification.
- Spam or Trash the comment. 
- Tap `Delete Permanently`.
- Verify:
  - An activity indicator is displayed on the button.
  - The comment notification is removed from the notifications list.
  - The first available notification is selected and the details are displayed.


| Button | Indicator |
|--------|-------|
| <img width="392" alt="light_button" src="https://user-images.githubusercontent.com/1816888/155253884-48390639-6486-44b1-a067-8be1b1c27f55.png"> | <img width="390" alt="light_indicator" src="https://user-images.githubusercontent.com/1816888/155253887-cd80dce5-5398-47dc-9f7f-f5c963e50c38.png"> |
| <img width="393" alt="dark_button" src="https://user-images.githubusercontent.com/1816888/155253891-e08f4c4b-2027-4723-81db-e7511d07a9a5.png"> | <img width="393" alt="dark_indicator" src="https://user-images.githubusercontent.com/1816888/155253889-784a71cd-7509-48ac-8f78-182d7687ff22.png"> |


## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
